subsys/dfu/dfu_target/src/dfu_target_suit.c

```diff
@@ -134,26 +170,66 @@ int dfu_target_suit_done(bool successful)
 
 int dfu_target_suit_schedule_update(int img_num)
 {
+	/* img_num is not needed for SUIT */
+	(void)img_num;
+
+	int err;
 	LOG_INF("Schedule update");
 
-	suit_plat_mreg_t update_candidate[] = {
-		{
-			.mem = DFU_PARTITION_ADDRESS,
-			.size = image_size,
-		},
-	};
+	/* Process the saved envelope */
+	err = suit_dfu_candidate_preprocess();
+	if (err != 0) {
+		LOG_ERR("suit_dfu_candidate_preprocess error %d", err);
+		return err;
+	}
+
+	return 0;
+}
 
-	suit_ssf_err_t err = suit_trigger_update(update_candidate, ARRAY_SIZE(update_candidate));
+int dfu_target_suit_reset(void)
+{
+	stream_buf_bytes = 0;
+	suit_ssf_err_t err;
+	int rc;
+
+	rc = dfu_target_stream_reset();
+	if (rc != 0) {
+		LOG_ERR("dfu_target_stream_reset error %d", rc);
+		return rc;
+	}
+
+	err = suit_dfu_cleanup();
+	stream_flash_in_use = false;
+	image_num = 0;
 
 	if (err != SUIT_PLAT_SUCCESS) {
-		return -EIO;
+		LOG_ERR("Failed to cleanup SUIT DFU");
+		rc = -EIO;
+	}
+
+	return rc;
+}
+
+int dfu_target_suit_reboot(void)
+{
+	suit_ssf_err_t err;
+
+	k_msleep(CONFIG_MGMT_SUITFU_TRIGGER_UPDATE_RESET_DELAY_MS);
+
+	err = suit_dfu_update_start();
+	if (err < 0) {
+		LOG_ERR("Failed to start firmware upgrade!");
+		return err;
```

Comment:
I can see you sometimes return prematurely on error and then return 0 below on success, and in other cases you just print error on failure, and return err at the end. It's a minor, but maybe it would be good to follow one pattern in all methods.

Reply:
Aligned. :)